File: app/src/main/java/com/mathocosta/learnbasic/activities/adapters/AllShoppingListsRecViewAdapter.java
package com.mathocosta.learnbasic.activities.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.mathocosta.learnbasic.R;
import com.mathocosta.learnbasic.models.ShoppingList;

import java.util.ArrayList;

public class AllShoppingListsRecViewAdapter
        extends RecyclerView.Adapter<AllShoppingListsRecViewAdapter.ViewHolder> {

    private ArrayList<ShoppingList> shoppingLists = new ArrayList<>();
    private final ShoppingListClickListener clickListener;

    public AllShoppingListsRecViewAdapter(ShoppingListClickListener clickListener) {
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.shopping_list_item, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bindView(shoppingLists.get(position), clickListener);
    }

    @Override
    public int getItemCount() {
        return shoppingLists.size();
    }

    public void setShoppingLists(ArrayList<ShoppingList> shoppingLists) {
        this.shoppingLists = shoppingLists;
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView txtName, txtCounter;
        private final CardView parent;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            txtName = itemView.findViewById(R.id.txt_name);
            txtCounter = itemView.findViewById(R.id.txt_counter);
            parent = itemView.findViewById(R.id.parent);
        }

        public void bindView(ShoppingList shoppingList, ShoppingListClickListener listener) {
            txtName.setText(shoppingList.getName());
            txtCounter.setText(
                    String.format(
                            "%d/%d",
                            shoppingList.getCheckedItems().size(),
                            shoppingList.getItems().size()
                    )
            );
            parent.setOnClickListener(view -> listener.onShoppingListClick(shoppingList));
        }
    }

    public interface ShoppingListClickListener {
        void onShoppingListClick(ShoppingList shoppingList);
    }
}

File: app/src/main/java/com/mathocosta/learnbasic/activities/adapters/AllShoppingListsRecViewAdapter.kt
package com.mathocosta.learnbasic.activities.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.mathocosta.learnbasic.R
import com.mathocosta.learnbasic.models.ShoppingList
import kotlinx.android.synthetic.main.shopping_list_item.view.*
import java.util.*

class AllShoppingListsRecViewAdapter(
        private val shoppingLists: MutableList<ShoppingList>,
        private val clickListener: ShoppingListClickListener
) : RecyclerView.Adapter<AllShoppingListsRecViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater
                .from(parent.context)
                .inflate(R.layout.shopping_list_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindView(shoppingLists[position], clickListener)
    }

    override fun getItemCount(): Int = shoppingLists.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val parent: CardView = itemView.findViewById(R.id.parent)

        fun bindView(shoppingList: ShoppingList, listener: ShoppingListClickListener) {
            itemView.txt_name.text = shoppingList.name

            val numberOfCheckedItems = shoppingList.checkedItems.size
            val numberOfItems = shoppingList.items.size
            itemView.txt_counter.text = "$numberOfCheckedItems/$numberOfItems"

            parent.setOnClickListener { listener.onShoppingListClick(shoppingList) }
        }

    }

    interface ShoppingListClickListener {
        fun onShoppingListClick(shoppingList: ShoppingList)
    }
}